### PR TITLE
@craigspaeth - Be defensive about checking for error on response object

### PR DESCRIPTION
**Pull request by broskoski**
Re: https://artsy.slack.com/archives/web/p1479207968001516

I'm guessing the request timed out (hence no response object), but that would be pretty hard to replicate / prove.

### CODE REVIEW

File: components/json_page/index.coffee [hunk 1]
```diff
@@ -30,13 +30,13 @@ module.exports = class JSONPage
         .end (err, res) =>
```
Comment (craigspaeth) [inline components/json_page/index.coffee]: I think this code wasn't updated when superagent changed its API to use the node-style error-first argument. We should handle `err` here instead.

Or bonus points if you drop the `.end` all-together and use superagent's new promise-based API.

Comment (craigspaeth) [inline components/json_page/index.coffee]: I think this code wasn't updated when superagent changed its API to use the node-style error-first argument. We should handle `err` here instead.

Or bonus points if you drop the `.end` all-together and use superagent's new promise-based API.

Comment (broskoski) [inline components/json_page/index.coffee]: Ah, good catch. Will update!

Comment (broskoski) [inline components/json_page/index.coffee]: Wait, confused, we are catching the error in the right order here. Am I missing something?

**Comment by joeyAghion** [comment]
Is it true that a failing request is hard to replicate in tests? I don't know, but I imagine there are solutions to this, and it could be a step in the direction we've been discussing in [#testing](https://artsy.slack.com/messages/testing/). #quality #tigerblood

**Comment by craigspaeth** [comment]
Replicating a failing request is not hard in unit tests because you can stub the response in superagent to fake an error for instance. Joey makes a good point we should probably write a test with this.

No good deed goes unpunished Cab 😬 Happy to follow up on your work and fill in a pending test too.

That said, for the sake of debate--bubbling up the error properly might be good enough for this component, and the test should be written further up where we would actually handle the error and display something useful to the user. Not sure how much value we get by writing tests against every potentially failing step of IO.

**Comment by broskoski** [comment]
No prob, I can add a test for this!

File: components/json_page/index.coffee (hunk 1 above)
Comment (craigspaeth) [inline components/json_page/index.coffee]: It's that we're handling 'res.error' instead or the 'err' object returned in the callback.

Comment (broskoski) [inline components/json_page/index.coffee]: Oh, yes. Duh.

Comment (broskoski) [inline components/json_page/index.coffee]: So based on http://visionmedia.github.io/superagent/#error-handling it looks like handling `res.error` is actually valid. I updated the code to be defensive about `res.error` and fall back to the `err.response`.

Comment (craigspaeth) [inline components/json_page/index.coffee]: 🤔  weird that there was no `res` object then

Comment (broskoski) [inline components/json_page/index.coffee]: Very weird!

**Pull request merged**